### PR TITLE
feat(menu): support extra prop

### PR DESCRIPTION
**Pull request by coding-ice**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [x] 🆕 New feature
- [ ] 🐞 Bug fix
- [x] 📝 Site / documentation improvement
- [x] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [x] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues
close https://github.com/ant-design/ant-design/issues/49127

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | menu support extra prop          |
| 🇨🇳 Chinese |  menu 支持 extra 属性         |

### CODE REVIEW

File: components/menu/style/index.ts [hunk 1]
```diff
@@ -635,13 +635,22 @@ const getBaseStyle: GenerateStyle<MenuToken> = (token) => {
...
           transition: `color ${motionDurationSlow}`,
 
           // https://github.com/ant-design/ant-design/issues/41143
           [`> ${antCls}-typography-ellipsis-single-line`]: {
             display: 'inline',
             verticalAlign: 'unset',
           },
+
+          [`${componentCls}-extra`]: {
+            marginLeft: 'auto',
```
Comment (li-jia-nan) [inline components/menu/style/index.ts:649]:
```suggestion
            marginInlineStart: 'auto',
```

File: package.json [hunk 2]
```diff
@@ -125,7 +125,7 @@
     "rc-input": "~1.6.3",
     "rc-input-number": "~9.2.0",
     "rc-mentions": "~2.15.0",
-    "rc-menu": "~9.14.1",
+    "rc-menu": "^9.15.0",
```
Comment (li-jia-nan) [inline package.json:128]:
```suggestion
    "rc-menu": "~9.15.0",
```

File: package.json [hunk 3]
```diff
@@ -125,7 +125,7 @@
     "rc-input": "~1.6.3",
     "rc-input-number": "~9.2.0",
     "rc-mentions": "~2.15.0",
-    "rc-menu": "~9.15.1",
+    "rc-menu": "^9.15.1",
```
Comment (yoyo837) [inline package.json:128]: 怎么又改回去了

Comment (coding-ice) [reply package.json]: 我看上面建议，是让我改掉，我改回来下

File: package.json [hunk 4]
```diff
@@ -125,7 +125,7 @@
     "rc-input": "~1.6.3",
     "rc-input-number": "~9.2.0",
     "rc-mentions": "~2.15.0",
-    "rc-menu": "~9.14.1",
+    "rc-menu": "~9.15.1",
```
Comment (li-jia-nan) [inline package.json:128]: 相关的库也需要提一下版本：

- #50455
- #50456
- #50457

**Comment by yoyo837** [comment]
image diff 单词多个s是本PR的原因？

**Comment by coding-ice** [comment]
> image diff 单词多个s是本PR的原因？

我看`diff`的更改，不涉及到这个，我更新下分支吧

**Comment by yoyo837** [comment]
最新的image diff 看上去 menu 的文字垂直方向有细微区别

**Comment by coding-ice** [comment]
> 最新的image diff 看上去 menu 的文字垂直方向有细微区别

这个是`v4`版本的，我没大看出来啥区别，大概率是`flex`垂直居中引起的，感觉可以忽略一下

**Comment by li-jia-nan** [comment]
@SocketSecurity ignore npm/@ant-design/tools@18.0.2
@SocketSecurity ignore npm/@biomejs/biome@1.8.3
@SocketSecurity ignore npm/puppeteer@22.15.0
@SocketSecurity ignore npm/sharp@0.33.5

**Review by li-jia-nan — APPROVED** [review]

**Review by yoyo837 — APPROVED** [review]

**Comment by yoyo837** [comment]
我来斗胆merge它。

File: components/menu/style/index.ts [hunk 5]
```diff
@@ -635,13 +635,26 @@ const getBaseStyle: GenerateStyle<MenuToken> = (token) => {
         },
 
         [`${componentCls}-title-content`]: {
+          display: 'inline-flex',
```
Comment (afc163) [inline components/menu/style/index.ts:638]: 导致了 https://github.com/ant-design/ant-design/issues/51361

**Pull request merged**